### PR TITLE
changed AWACS orbit to a racetrack

### PR DESCRIPTION
**Pull request by Raskil**
This PR will change the AWACS orbit to race tracks. During the circular orbit the awacs flies a steep bank angle. This will result in periodical data link loss. The long racetrack pattern should prevent this.

### CODE REVIEW

**Comment by jrcnn** [comment]
A bigger orbit radius would also fix this, and would make for a more predictable flight pattern. A circle is also more realistic.

**Comment by Raskil** [comment]
> A bigger orbit radius would also fix this, and would make for a more predictable flight pattern. A circle is also more realistic.

I did not find a way to specify the orbit radius for an circular orbit way point in dcs.

**Comment by jrcnn** [comment]
> 
A normal orbit waypoint will always make the airplane do a 30° bank.

However, I believe the orbit function uses the next waypoint to define the diameter of the circle.

**Comment by DanAlbert** [comment]
It doesn't. If it did some of these orbits would be hundreds of nm across. That's just how racetracks work.

The orbit speed might indirectly control the diameter of the orbit, but afaik there's no direct control.

**Comment by DanAlbert** [comment]
Does adjusting the speed of the orbit adjust the diameter?

**Comment by jrcnn** [comment]
> Does adjusting the speed of the orbit adjust the diameter?

Yes, but it doesn't matter because the bank angle stays the same, so the data loss will still happen.

**Comment by DanAlbert** [comment]
So is this a workaround for a DCS bug then? Has that been reported to ED?

**Comment by jrcnn** [comment]
> So is this a workaround for a DCS bug then? Has that been reported to ED?

What bug? I don't see how this fixes anything.

**Comment by DanAlbert** [comment]
The bug that AWACS aircraft drop data link when in an orbit. I've only seen AWACS on a circular orbit in any mission I've played. If that's dropping data link because the aircraft are banking too hard and there's nothing mission editors can do to fix that other than switching to a racetrack, it sounds like a bug.

**Comment by SnappyComebacks** [comment]
> it sounds like a bug.

BigNewy says it is not a bug, so switching to a racetrack must be the way to go unless that changes.
https://forums.eagle.ru/topic/281416-awacs-blindstops-when-orbiting/

**Comment by DanAlbert** [comment]
Ah, so it's a radar blind spot, not a data link disconnection. Okay, that makes more sense. Will review this later then.

**Comment by DanAlbert** [comment]
(please do add a comment linking to that thread though for whomever ends up reading that code next though 🙂)

File: gen/flights/flightplan.py [hunk 1]
```diff
@@ -1195,39 +1189,79 @@ def generate_strike(self, flight: Flight) -> StrikeFlightPlan:
...
-    def generate_aewc(self, flight: Flight) -> AwacsFlightPlan:
-        """Generate a AWACS flight at a given location.
+    def generate_aewc_racetrack(self, flight: Flight) -> AwacsFlightPlan:
+        """Generate a AWACS flight in racetrack orbit.
 
         Args:
             flight: The flight to generate the flight plan for.
         """
+
+        racetrack_half_distance = Distance.from_nautical_miles(20).meters
```
Comment (SnappyComebacks) [inline gen/flights/flightplan.py]:
```suggestion
        racetrack_half_distance = Distance.from_nautical_miles(30).meters
```

AWACS should probably have longer legs to turn less often.

**Comment by Raskil** [comment]
Thx guys for accepting and improving. That bug really bothers me a lot :)

**Comment by Raskil** [comment]
> The orbit speed might indirectly control the diameter of the orbit, but afaik there's no direct control.

@DanAlbert Orbit is an equotation between speed (Groundspeed) bank angle and orbit radius. So if Bank angle and speed are set (which they are for an orbit waypoint in DCS), there is no possible way to set radius as well.
If you ever need the calculations, you can grab them here: [Turn Radius Discussion](https://code7700.com/rot_turn_radius.htm)
I once made an google docs sheet to calculate stuff around orbits (radius, bank angle and speed): [Spreadsheet](https://docs.google.com/spreadsheets/d/1qVtrWxSllj8GcsSLAXG-MSq_5tLoaMJo7BBj2SXoor0/edit#gid=1465831768)

**Pull request merged**